lib/cisco_node_utils/cmd_ref/README_YAML.md

```diff
@@ -0,0 +1,457 @@
+
+Any of the attribute properties can be subdivided by client API using the
+API name (in lower case) as a key. For example, interface VRF membership defaults to "" (no VRF) in both NXAPI and gRPC clients, but the CLI is 'vrf member <vrf>' for NXAPI and 'vrf <vrf>' for gRPC. Thus, the YAML could be written as:
+
+```yaml
+# interface.yaml
+vrf:
+  default_value: ""
+  nxapi:
+    config_get_token_append: '/^vrf member (.*)/'
+    config_set_append: "<state> vrf member <vrf>"
+  grpc:
+    config_get_token_append: '/^vrf (.*)/'
+    config_set_append: "<state> vrf <vrf>"
+```
+
+### Platform variants
+
+Any of the attribute properties can be subdivided by platform product ID string
+using a regexp against the product ID as a key. When one or more regexp keys
+are defined thus, you can also use the special key `else` to provide values
+for all products that do not match any of the given regexps:
+
+```yaml
+# show_version.yaml
+system_image:
+  /N9K/:
+    config_get_token: "kick_file_name"
+    test_config_get_regex: !ruby/regexp '/.*NXOS image file is: (.*)$.*/'
+  else:
+    config_get_token: "isan_file_name"
+    test_config_get_regex: !ruby/regexp '/.*system image file is:    (.*)$.*/'
+```
+
+### Combinations of these
+
+In many cases, supporting multiple APIs and multiple platforms will require
+using several or all of the above options.
+
+Using `_template` in combination with API variants:
+
+```yaml
+# inventory.yaml
+_template:
+  grpc:
+    config_get: 'show inventory | begin "Rack 0"'
+    test_config_get: 'show inventory'
+  nxapi:
+    config_get: 'show inventory'
+    test_config_get: 'show inventory | no-more'
+
+productid:
+  grpc:
+    config_get_token: '/PID: ([^ ,]+)/'
+  nxapi:
+    config_get_token: ["TABLE_inv", "ROW_inv", 0, "productid"]
+```
+
+Using API variants and platform variants together:
+
+```yaml
+# inventory.yaml
+description:
+  config_get_token: "chassis_id"
+  nxapi:
+    /N7K/:
+      test_config_get_regex: !ruby/regexp '/.*Hardware\n  cisco (\w+ \w+ \(\w+ \w+\) \w+).*/'
+    else:
+      test_config_get_regex: !ruby/regexp '/Hardware\n  cisco (([^(\n]+|\(\d+ Slot\))+\w+)/'
+  grpc:
+    config_get: 'show inventory | inc "Rack 0"'
+    config_get_token: '/DESCR: "(.*)"/'
+    test_config_get: 'show inventory | inc "Rack 0"'
+    test_config_get_regex: !ruby/regexp '/DESCR: "(.*)"/'
+```
+
+## Attribute properties
+
+### `config_get`
+
+`config_get` must be a single string representing the CLI command (usually a
+`show` command) to be used to display the information needed to get the
+current value of this attribute.
+
+```yaml
+# interface_ospf.yaml
+area:
+  config_get: 'show running interface all'
+```
+
+### `config_get_token`
+
+`config_get_token` can be a single string, a single regex, an array of strings,
+or an array of regexs.
+
+If this value is a string or array of strings, then the `config_get` command
+will be executed to produce _structured_ output and the string(s) will be
```

Comment:
We should probably add a word of caution to make sure that _structured_ output works across all target platforms.